src/parse.rs

```diff
@@ -85,7 +87,9 @@ pub fn parse_module(src: &str) -> Result<(UntypedModule, ModuleExtra), ParseErro
 pub fn parse_expression_sequence(src: &str) -> Result<UntypedExpr, ParseError> {
     let lex = lexer::make_tokenizer(src);
     let mut parser = Parser::new(lex);
-    if let Some((e, _)) = parser.parse_expression_seq()? {
+    let expr = parser.parse_expression_seq();
+    let expr = parser.ensure_no_errors(expr)?;
+    if let Some((e, _)) = expr {
```

Comment:
pass parse_expression_sequence through the same error process as parse_module so we can get at LexicalErrors